cluster/juju/layers/kubernetes-worker/reactive/kubernetes_worker.py

```diff
@@ -25,6 +26,12 @@
 def remove_installed_state():
     remove_state('kubernetes-worker.components.installed')
 
+@hook('stop')
+def remove_this_node():
+    ''' Deletes this unit's node when the unit is destroyed. '''
+    kubectl = ['kubectl', '--kubeconfig=/srv/kubernetes/config']
+    command = kubectl + ['delete', 'node', gethostname()]
+    output = check_call(command, shell=False)
```

Comment:
@mbruzek would it make more sense to add a 'shutting-down' state, then use `@when_not('shutting-down')` everywhere we have a `@when('kubernetes-worker.components.installed')`?